src/libs/antares/study/variable-print-info.cpp

```diff
@@ -83,120 +82,120 @@ void variablePrintInfoCollector::add(const AnyString& name,
                                      uint dataLevel,
                                      uint fileLevel)
 {
-    allvarsinfo->add(new VariablePrintInfo(name, dataLevel, fileLevel));
+    allvarsinfo->add(name.to<std::string>(), VariablePrintInfo(dataLevel, fileLevel));
 }
 
 // ============================================================
 // All variables print information
 // ============================================================
 
-AllVariablesPrintInfo::~AllVariablesPrintInfo()
+static std::string to_uppercase(std::string str)
 {
-    clear();
+    std::transform(str.begin(), str.end(), str.begin(), ::toupper);
+    return str;
 }
 
-void AllVariablesPrintInfo::add(VariablePrintInfo* v)
+void AllVariablesPrintInfo::add(std::string name, VariablePrintInfo v)
 {
-    allVarsPrintInfo.push_back(v);
+    std::string upperCaseName = to_uppercase(name);
+    if (not exists(upperCaseName))
+    {
+        index_to_name[(unsigned int)allVarsPrintInfo.size()] = upperCaseName;
+        allVarsPrintInfo.insert(std::pair<std::string, VariablePrintInfo>(upperCaseName, v));
```

Comment:
About that, I first wrote the line : 
```C++
allVarsPrintInfo[upperCaseName] = v;
```
But it does not compile if class **VariablePrintInfo** (type of the values in the **std::map**) does not have an explicitly defined **default constructor**.
Indeed, as you may know, the **std::map [] operator** needs to add a default value in the map in case the key does have already a associated value.
As I didn't want this **default constructor** to be defined (to make code shorter), I opted for the **insert** method.